### PR TITLE
Add more extensions

### PR DESCRIPTION
**Pull request by zoggy**
This adds a `more-mime.types` file with additional extensions, so that the "official" ones are in `mime.types`.

`generate_mime_types.ml` now reads on stdin if no argument is given.

`generate.sh` now sends `mime.types` and `more-mime.types` to generate_mime_types.ml so that official extensions are completed with the ones from `more-mime.types`.

### CODE REVIEW

**Comment by zoggy** [comment]
For information, I took additional extension -> mime type mapping from gtksourceview lang files.

File: more-mime.types [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+text/x-forth		frt fs
+text/x-fortran	f f90 f95 for
+text/x-fsharp		fs
+text/x-gap			g gd gi gap
+text/x-gtkrc		gtkrc
+text/x-literate-haskell		lhs
+text/x-haskell						hs
+text/x-idl				idl
+text/x-ini-file		ini
+text/x-javascript		js
```
Comment (dsheets) [inline more-mime.types]: I don't think this is standard...

Comment (zoggy) [inline more-mime.types]: I removed it. Anyway, it was not in the final mime_types.ml file, as js is already associated in mime.types.

**Comment by zoggy** [comment]
I commited a change to handle some filenames with no extension. I wonder whether the filename should be lowercased before lookup or not.
The filenames handled are defined in `files.types` and the code is generated by `generate_mime_types.ml` when given the `--files` option.

**Comment by dsheets** [comment]
`application/font-woff`

**Comment by samoht** [comment]
@dsheets any opinion on the current state of that PR?

**Comment by dsheets** [comment]
I don't think media type assumption from file name extensions is a good idea in general. This library continues to be misnamed without any `file`-like detection component.

These changes looks reasonable to me. I'm not sure about the organization of the database of media types <-> name extensions. In particular, why are all of the unofficial programming language `x-` types in a file called `more-mime.types`? Why not `pl-x-mime.types`? Also, why not add `application/font-woff` to the primary database? If the primary database is tracking some external source, then additional, official, auxiliary mappings should have their own file. Anyway, these are not blockers and I would merge this as-is.

**Comment by djs55** [comment]
I couldn't quite tell whether these changes were good to merge as-is and I ended up making conflicting changes -- sorry! Let me know if you'd like me to rebase them for you.

**Comment by avsm** [comment]
@zoggy I'm very sorry about the multi-year gap in addressing your PR, but I've now cut a release :-)  To address @dsheets' comments, I have documented how the original mime.types was obtained (from Apache SVN) and also made the `x-mime.types` filename clearer.

**Pull request merged**